Surveys/Sources/Presentation/Modules/Survey/SurveyView.swift

```diff
@@ -0,0 +1,103 @@
+//
+//  SurveyView.swift
+//  Surveys
+//
+//  Created by David Bui on 28/12/2022.
+//  Copyright © 2022 Nimble. All rights reserved.
+//
+
+import Kingfisher
+import SwiftUI
+
+struct SurveyView: View {
+
+    @EnvironmentObject private var navigator: Navigator
+    @ObservedObject private var viewModel: SurveyViewModel
+
+    var body: some View {
+        switch viewModel.state {
+        case .idle, .loaded:
+            setUpView()
+        case .loading:
+            setUpView(isLoading: true)
+        case let .error(message):
+            setUpView()
+                .alert(isPresented: .constant(true)) {
+                    Alert(
+                        title: Text(Localize.commonApplicationNameText()),
+                        message: Text(message),
+                        dismissButton: Alert.Button.default(Text(Localize.commonOkText()))
+                    )
+                }
+        }
+    }
+
+    init(viewModel: SurveyViewModel) {
+        self.viewModel = viewModel
+    }
+
+    private func setUpView(isLoading: Bool = false) -> some View {
+        ZStack {
+            setUpBackground()
+            setUpSurvey()
+        }
+        .modifier(NavigationBackButtonModifier(action: {
+            navigator.goBack()
+        }))
+    }
+
+    private func setUpBackground() -> some View {
+        KFImage(URL(string: viewModel.survey.coverImageUrl))
+            .placeholder { _ in
+                Assets.surveyBackgroundImage.image
+                    .resizable()
+                    .scaledToFill()
+            }
+            .fade(duration: 0.3)
+            .resizable()
+            .scaledToFill()
+            .overlay {
+                Constants.Gradient.surveyBackground.opacity(0.6)
+            }
+            .frame(minWidth: 0.0, maxWidth: .infinity)
```

Comment:
do we need `minWidth`?
```suggestion
            .frame(maxWidth: .infinity)
```

Reply:
Yes, otherwise, the UI will be like this

<img src=https://user-images.githubusercontent.com/24598204/209900732-7f5ce426-6728-41c0-9e68-f3dc4fb37259.png width=400>

Reply:
I just tested setting the frame with geo, and it worked. I will improve this in #16 task